"do while loop is not stopping even though the condition no longer met i have a do while loop whose condition depends on a property value of an object. the object looks like this:..public class myobjecttype.{.   public int id { get; set; }.   public bool isvalid { get; set; }.   public datetime completiontimestamp { get; set; }.   public string comments { get; set; }.}...the do while loop looks like this:..public task dosomething(myobjecttype myobject).{.   do.   {.      // call db get some values.      var somedata = _myrepository.get(myobject.id);..      // process data.      somedata.somevalue = somefunction(somedata.property1, somedata.property2);..      // not pretty code but i'm trying to set a few values for myobject here.      if(somedata.somevalue == null).         myutils.populateerrordata(ref myobject, ""something went wrong!!!"");..      // save data -- when there's an error, i don't want to this this step.      await _myrepository.save(somedata);.   }.   while(myobject.isvalid);.}...the populateerrordata() function is just a place where i set the values of myobject:..public static void populateerrordata(ref myobjecttype myobject, string comments).{.    myobject.isvalid = false;.    myobject.completiontimestamp = datetime.utcnow;.    myobject.comments = comments;.}...i can't say that i'm really happy about this approach but in my dosomething() function which is a backend maintenance function, i need to go through a lot of steps and if any of them fails, i want to abort the process...i figured i can just use a do while and when the condition is no longer met, it would kick me out of the loop anyway but it is not. it just keeps on going. in the above code, if somedata.somevalue == null, i should never save the data but as i said, the code continues and i end up saving data in the database...any idea why it doesn't kick me out of the loop even though myobject.isvalid is set to false by populateerrordata()?"
